launcher/java/JavaUtils.cpp

```diff
@@ -202,14 +202,8 @@ QList<JavaInstallPtr> JavaUtils::FindJavaFromRegistryKey(DWORD keyType, QString
     {
         // Read the current type version from the registry.
         // This will be used to find any key that contains the JavaHome value.
-        char *value = new char[0];
+        TCHAR *value = new TCHAR[0];
         DWORD valueSz = 0;
```

Comment:
Why allocate an empty buffer? Also, why does the code use a TCHAR string if assumes it's always gonna be a WCHAR string? Also, why declare it in such a wide scope when it's never used in that scope?
```suggestion
```